p2p/node/dhtstorage/datastore.go

```diff
@@ -0,0 +1,156 @@
+package dhtstorage
+
+import (
+	"context"
+
+	"github.com/dominant-strategies/go-quai/log"
+	datastore "github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+// ipfs Datastore interface implementation
+
+// ensure DHTStorage implements the datastore interface
+var _ datastore.Datastore = (*DHTStorage)(nil)
+
+// Get retrieves the object `value` named by `key`.
+// Get will return ErrNotFound if the key is not mapped to a value.
+func (ds *DHTStorage) Get(_ context.Context, key datastore.Key) (value []byte, err error) {
```

Comment:
Can you please add some tests or more importantly just make sure that Get and Put work correctly in a live environment when dealing with PeerIDs. These issues can be very difficult to debug later.

Reply:
Can you verify in a live environment? The last peerdb PR had unit tests but they didn't capture issues that happened in the live environment

Reply:
I didn't realize that the DHT being added here isn't being utilized. In my opinion the PR should implement the complete feature of storing *and* bootstrapping from the dht.